neo/core/spiketrainlist.py

```diff
@@ -0,0 +1,294 @@
+                self._items = None
+                return self
+            else:
+                return self.__class__.from_spike_time_array(
+                    combined_spike_time_array,
+                    combined_channel_id_array,
+                    combined_channel_ids,
+                    t_start=self._spiketrain_metadata['t_start'],
+                    t_stop=self._spiketrain_metadata['t_stop'])
+
+    def __add__(self, other):
+        """Return self + other"""
+        if isinstance(other, self.__class__):
+            return self._add_spiketrainlists(other)
+        elif other and is_spiketrain_or_proxy(other[0]):
+            return self._add_spiketrainlists(
+                self.__class__(items=other, segment=self.segment)
+            )
+        else:
+            if self._items is None:
+                self._spiketrains_from_array()
+            return self._items + other
+
+    def __iadd__(self, other):
+        """Return self"""
+        if isinstance(other, self.__class__):
+            return self._add_spiketrainlists(other, in_place=True)
+        elif other and is_spiketrain_or_proxy(other[0]):
+            for obj in other:
+                obj.segment = self.segment
+            if self._items is None:
+                self._spiketrains_from_array()
+            self._items.extend(other)
+            return self
+        else:
+            raise TypeError("Can only add a SpikeTrainList or a list of SpikeTrains in place")
+
+    def __radd__(self, other):
+        """Return other + self"""
+        if isinstance(other, self.__class__):
+            return other._add_spiketrainlists(self)
+        elif other and is_spiketrain_or_proxy(other[0]):
+            for obj in other:
+                obj.segment = self.segment
+            if self._items is None:
+                self._spiketrains_from_array()
+            self._items.extend(other)
+            return self
+        elif len(other) == 0:
+            return self
+        else:
+            if self._items is None:
+                self._spiketrains_from_array()
+            return other + self._items
+
+    def append(self, obj):
+        """L.append(object) -> None -- append object to end"""
+        if not is_spiketrain_or_proxy(obj):
+            raise ValueError("Can only append SpikeTrain objects")
+        if self._items is None:
+            self._spiketrains_from_array()
+        obj.segment = self.segment
+        self._items.append(obj)
+
+    def extend(self, iterable):
+        """L.extend(iterable) -> None -- extend list by appending elements from the iterable"""
+        if self._items is None:
+            self._spiketrains_from_array()
+        for obj in iterable:
+            obj.segment = self.segment
+        self._items.extend(iterable)
+
+    @classmethod
+    def from_spike_time_array(cls, spike_time_array, channel_id_array,
+                              all_channel_ids=None, units=None,
```

Comment:
I think that all_channel_ids should be a necessary arguments.
Making `unique()` is dangerous when dealing with smal chunks a  neuron can have no spike in one segment.

Reply:
done.

Reply:
so the difference is that `channel_id_array` has the same dimension as `spike_time_array` and `all_channel_ids` has to contain all unique entries present in `channel_id_array` plus some additional ones if a neuron didn't spike? Maybe it would be good to add an argument description in the docstring to avoid confusion....